### PR TITLE
Added support for oracle database

### PR DESCRIPTION
**Pull request by ElectricVampire**
I did try to add support for oracle, if its ok - I can easily extend this for mysql as well.

I know execution plan looks pretty straight forward without any interactive UI like SQL server, but this is how I see it today in Sql Developer(IDE for Oracle) as well, so for start I think it's good as it will help - I am working on making it as tree but couple of my friends who are oracle DBA from multiple years don't like it as they are used of the view which I have updated in readme.

Thank you for great work with this extension. Let me know your thoughts on this PR

### CODE REVIEW

**Comment by Giorgi** [comment]
Can you post a screenshot of what it looks like?

**Comment by ElectricVampire** [comment]
I have added in readme as well you check out in PR file diff or you can check it out here
https://github.com/ElectricVampire/EFCore.Visualizer/tree/master?tab=readme-ov-file#oracle

File: src/IQueryableObjectSource/DatabaseProvider.cs [hunk 1]
```diff
@@ -83,4 +83,24 @@ protected override string ExtractPlanInternal(DbCommand command)
...
+
+class OracleDatabaseProvider(DbCommand command) : DatabaseProvider(command)
+{
+    protected override string ExtractPlanInternal(DbCommand command)
+    {
+        command.CommandText = "EXPLAIN PLAN FOR " + command.CommandText;
+        command.ExecuteNonQuery();
+
+        // Querying the execution plan using DBMS_XPLAN
+        command.CommandText = "SELECT * FROM TABLE(DBMS_XPLAN.DISPLAY())";
```
Comment (Giorgi) [inline src/IQueryableObjectSource/DatabaseProvider.cs:96]: Does this give an actual query plan or an estimated one?

Comment (ElectricVampire) [reply src/IQueryableObjectSource/DatabaseProvider.cs:96]: Actual Plan

Comment (ElectricVampire) [reply src/IQueryableObjectSource/DatabaseProvider.cs:96]: This is an estimated plan, should we give option for actual plan or show only actual plan

Comment (Giorgi) [reply src/IQueryableObjectSource/DatabaseProvider.cs:96]: Actual is better as that's how it behaves for other databases.

Comment (ElectricVampire) [reply src/IQueryableObjectSource/DatabaseProvider.cs]: > Actual is better as that's how it behaves for other databases.

Done, also added a line in readme which make it clear to users that all plans are actual query plans

Comment (ElectricVampire) [reply src/IQueryableObjectSource/DatabaseProvider.cs]: Can you please also tell me scope for this PR, would you consider it only when tree visuals are ready?

Comment (Giorgi) [reply src/IQueryableObjectSource/DatabaseProvider.cs]: Textual plan is fine. You can do the tree in a separate PR if you want to. I'll review the changes later today.

Comment (ElectricVampire) [reply src/IQueryableObjectSource/DatabaseProvider.cs]: > Textual plan is fine. You can do the tree in a separate PR if you want to. I'll review the changes later today.

Thanks, I shared this with my oracle DBA friends they liked the plan content specially I/O stats, but they complained hitting 5 sec time out more often than not when I looked its mostly in first connection and specially when its remote DB in another region - Like their machine is in ASIA and they are connecting DB in US East.
I see in known issues, need to think how we can get around this.
Just throwing an idea - Few weeks ago I wrote an extension to do Sonar scan in VS with Sonar Qube, To do this I used docked browser control at bottom of VS, Do you think its worth exploring this path?

File: src/IQueryableObjectSource/DatabaseProvider.cs [hunk 2]
```diff
@@ -83,4 +83,36 @@ protected override string ExtractPlanInternal(DbCommand command)
...
+    {
+        using var statisticsCommand = command.Connection.CreateCommand();
+        try
+        {
+            statisticsCommand.Transaction = command.Transaction;
+            statisticsCommand.CommandText = "ALTER SESSION SET statistics_level = ALL";
+            statisticsCommand.ExecuteNonQuery();
+
+            // We need empty the reader stream, so V$SQL_PLAN has all the stats, otherwise when we will query the plan - we will get older plan
+            var res = command.ExecuteReader();
```
Comment (Giorgi) [inline src/IQueryableObjectSource/DatabaseProvider.cs:100]: You need to `Dispose` the `res` variable

**Pull request merged**